internal/provider/resource_org_member.go

```diff
@@ -207,6 +209,27 @@ func (r *OrgMemberResource) Create(ctx context.Context, req resource.CreateReque
 		if invite.Invite.ID != "" {
 			data.InviteID = types.StringValue(invite.Invite.ID)
 		}
+
+		// If username or email or not set, we need to compute them
+		if data.UserName.ValueString() == "" {
+			data.UserName = types.StringValue("")
+		}
+		if data.Email.ValueString() == "" {
+			data.Email = types.StringValue("")
+		}
+	}
+
+	if current.InviteID.ValueString() == "" &&
+		current.Role.ValueString() != data.Role.ValueString() {
```

Comment:
```suggestion
    isAlreadyMember := current.InviteID.ValueString() == ""
    isRoleUpdated := current.Role.ValueString() != data.Role.ValueString()

	if isAlreadyMember && isRoleUpdated {
```

Just some small changes for readability. I can go either way, but before this change it took some time of thinking on the code to think about what `current.InviteID.ValueString == ""` meant

Reply:
i like that!